app/controllers/posts_controller.rb

```diff
@@ -12,4 +12,34 @@ def show
 
     redirect_to user_posts_path(@user)
   end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+
+    if @post.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = @post.likes.build(user: current_user)
+
+    if @like.save
+      redirect_to user_post_path(@post.user, @post)
+    else
+      render :like
+    end
+  end
+
```

Comment:
[REQUIRED] For now kindly stick to the main actions like index, create, show and others. If you need to create something for likes then create an action in the likes controller then create a relationship between posts and likes. So kindly remove this.

Reply:
Well resolved ✔️